src/ctags.d

```diff
@@ -28,101 +30,143 @@ void printCtags(File output, string[] fileNames)
 	foreach (fileName; fileNames)
 	{
 		File f = File(fileName);
-		if (f.size == 0) continue;
+		if (f.size == 0)
+			continue;
 		auto bytes = uninitializedArray!(ubyte[])(to!size_t(f.size));
 		f.rawRead(bytes);
 		auto tokens = getTokensForParser(bytes, config, &cache);
 		Module m = parseModule(tokens.array, fileName, null, &doNothing);
+
 		auto printer = new CTagsPrinter;
 		printer.fileName = fileName;
 		printer.visit(m);
 		tags ~= printer.tagLines;
 	}
-	output.write("!_TAG_FILE_FORMAT\t2\n"
-			~ "!_TAG_FILE_SORTED\t1\n"
-			~ "!_TAG_FILE_AUTHOR\tBrian Schott\n"
-			~ "!_TAG_PROGRAM_URL\thttps://github.com/Hackerpilot/Dscanner/\n");
+	output.write(
+		"!_TAG_FILE_FORMAT\t2\n" ~ "!_TAG_FILE_SORTED\t1\n" ~ "!_TAG_FILE_AUTHOR\tBrian Schott\n" ~ "!_TAG_PROGRAM_URL\thttps://github.com/Hackerpilot/Dscanner/\n");
 	tags.sort().copy(output.lockingTextWriter);
 }
 
 private:
 
-void doNothing(string, size_t, size_t, string, bool) {}
+/// States determining how an access modifier affects the parsning.
+/// Reset, when ascending the AST reset back to the previous access.
+/// Keep, when ascending the AST keep the new access.
+enum AccessState
+{
+	Reset,
+	Keep
+}
+
+alias ContextType = Tuple!(string, "c", string, "access");
+
+void doNothing(string, size_t, size_t, string, bool)
+{
+}
+
+string paramsToString(Dec)(const Dec dec)
+{
+	import std.d.formatter : Formatter;
+
+	auto app = appender!string();
+	auto formatter = new Formatter!(typeof(app))(app);
+
+	static if (is(Dec == FunctionDeclaration))
```

Comment:
You can write this as `is(Dec == FunctionDeclaration) || is(Dec == Constructor)` and then delete lines 82-85.